examples/qtmegachatapi/chatWindow.cpp

```diff
@@ -690,3 +696,37 @@ void ChatWindow::deleteCallGui()
     ui->mTextChatWidget->show();
 }
 #endif
+
+void ChatWindow::on_mAttachBtn_clicked()
+{
+    QString node = QFileDialog::getOpenFileName(this, tr("All Files (*)"));
+
+    if (node.isEmpty())
+       return;
+
+    QStringList nodeParsed = node.split( "/" );
+    QString nodeName = nodeParsed.value(nodeParsed.length() - 1);
+    mega::MegaNode *parent = mMegaApi->getNodeByPath("/");
+    QMessageBox::warning(nullptr, tr("Node attachment"), tr("Please don't close the window until the transfer has been finished "));
```

Comment:
I'll prepare a modal dialog and close it once the upload completes, allowing to cancel in the meantime.